src/VirtoCommerce.PricingModule.Web/Scripts/blades/assignment-detail.js

```diff
@@ -141,9 +146,80 @@ angular.module('virtoCommerce.pricingModule')
     };
 
     // actions on load
-    $scope.catalogs = catalogs.query();
-    pricelists.search({ take: 1000 }, function (result) {
-    	$scope.pricelists = result.results;
-    });
+    $scope.fetchCatalogs = ($select) => {
+        while (blade.isLoading) sleep(100);
+        if ($scope.catalogs.length == 0) {
+            $select.page = 0;
+
+            if (blade.currentEntity.catalogId) {
+                let criteria = {
+                    CatalogIds: [blade.currentEntity.catalogId]
+                }
+                catalogs.search(criteria, (data) => {
+                    $scope.catalogs = data.results;
+                    $scope.fetchNextCatalogs($select);
+                });
+            }
+            else {
+                $scope.fetchNextCatalogs($select);
+            }
+        }
+    }
+    $scope.fetchNextCatalogs = ($select) => {
+        let criteria = {
+            SearchPhrase: $select.search,
+            take: $scope.pageSize,
+            skip: $select.page * $scope.pageSize
+        }
+
+        catalogs.search(criteria, (data) => {
+            $scope.catalogs = $scope.catalogs.concat(data.results);
+            $select.page++;
```

Comment:
can it both search and prevent duplicates?
And searches without skipping the results?